### ISSUE TITLE
Accessibility issues with floating labels/validation

### ISSUE THREAD

**Issue by HaoK** [twbs/bootstrap#35093]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

We are using the floating label look (https://getbootstrap.com/docs/5.1/forms/floating-labels/) for our forms and we've run into an accessibility issue where in High Contrast mode, users expected that the fields and text inside should be in proper orientation since this may cause confusion for our keyboard users

### Reduced test cases

Validation example in the docs demonstrates the issue (https://getbootstrap.com/docs/5.1/forms/floating-labels/)

![image](https://user-images.githubusercontent.com/6537861/135663579-f1a91be1-b267-4d74-a5af-31846c7918ec.png)

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.1.0

**Comment by Mohamed4salatin** [comment]
<body><liste>
<form class="email">

**Comment by merriemcgaw** [comment]
@mdo I believe the issue is that the boundaries of the label fall outside the boundaries of the text box.

**Comment by scottaohara** [comment]
Here's a demonstration of what is happening when someone has turned on windows high contrast mode:
<img width="549" alt="high contrast mode label has border and overlap problems with placeholder text" src="https://user-images.githubusercontent.com/4152514/140321233-8bb3fcb3-e6f9-4703-8044-a2df7a8b3908.PNG">

colors that are defined as transparent become visible in high contrast mode, hence the above screenshot where the label's transparent borders, and the transparent placeholder text appear and create a poor user experience.

These styles should be adjusted either by refining the styling to remove these bad side effects for [windows high contrast mode](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/#:~:text=New%2C%20standard%20system%20colors%20replace%20the%20old%20values%3A,Highlight%20..%20...%20%202%20more%20rows%20), but ideally the functionality of the positioning / dependency on the `placeholder` text should be re-considered.  That way, authors wouldn't have to be informed of / have to worry about any styling updates they make having to work in both standard and high contrast modes, as well as the fact that requiring the use of placeholder text means that screen readers will announce, likely unnecessary, extra content for each of these form fields.

cc @patrickhlauke as an fyi

**Comment by patrickhlauke** [comment]
still trying to understand what the original problem reported by @HaoK is - not sure if @scottaohara's point is something else/unrelated or not.

as an aside, note that for security, some browsers (like Chrome) don't report/apply the `:placeholder-shown` pseudo until after a user has interacted with a control (presumably to avoid inferring information silently from a user, e.g. "has visited the page before", using some hidden form field)

**Comment by scottaohara** [comment]
My comment was to add clarity of what the issue is actually about.

**Comment by patrickhlauke** [comment]
so "users expected that the fields and text inside should be in proper orientation since this may cause confusion for our keyboard users" is meant to mean "in WHCM the floated label covers up the placeholder/text, then?

(might have been more immediately apparent if the screenshot had been of the same thing, or maybe i just haven't had enough coffee yet today)

**Comment by HaoK** [comment]
Yeah @scottaohara summarizes some of the issues well, sorry for the unclear description

**Comment by mdo** [comment]
Closing as dupe of other issue for WHCM issues.
